test/commands/test_get.py

```diff
@@ -1374,8 +1038,336 @@ def test_get_workflow_version_history_json(
+        )
+
+    def _test_get_workflow_instances_with_datetime_filter(
+        self,
+        filter_argument: str,
+        mock_datetime_filter,
+        json: bool,
+    ):
+        """Helper method for testing that the 'get workflow-instances' command
+        works correctly with a given datetime filter"""
+
+        # SETUP
+        date = datetime(2023, 1, 1)
+
+        # CALL & ASSERT
+        self._test_get_workflow_instances_with_filters(
+            filter_arguments=[filter_argument, date.strftime(DATE_INPUT_FORMAT)],
+            expected_filters=[mock_datetime_filter.return_value],
+            json=json,
+        )
+        mock_datetime_filter.assert_called_once_with(date)
+
+    def test_get_workflow_instances_with_start_filter(
+        self,
+    ):
+        """Tests that the 'get workflow-instances' command works correctly
+        (while filtering by start date/time)"""
+
+        self._test_get_workflow_instances_with_datetime_filter(
+            filter_argument="--start",
+            mock_datetime_filter=self.mock_start_filter,
+            json=False,
+        )
+
+    def test_get_workflow_instances_with_start_filter_json(
+        self,
+    ):
+        """Tests that the 'get workflow-instances' command works correctly
+        (while filtering by start date/time and json=True)"""
+
+        self._test_get_workflow_instances_with_datetime_filter(
+            filter_argument="--start",
+            mock_datetime_filter=self.mock_start_filter,
+            json=True,
+        )
+
+    def test_get_workflow_instances_with_end_filter(
+        self,
+    ):
+        """Tests that the 'get workflow-instances' command works correctly
+        (while filtering by end date/time)"""
+
+        self._test_get_workflow_instances_with_datetime_filter(
+            filter_argument="--end",
+            mock_datetime_filter=self.mock_end_filter,
+            json=False,
+        )
+
+    def test_get_workflow_instances_with_end_filter_json(
+        self,
+    ):
+        """Tests that the 'get workflow-instances' command works correctly
+        (while filtering by end date/time and json=True)"""
+
+        self._test_get_workflow_instances_with_datetime_filter(
+            filter_argument="--end",
+            mock_datetime_filter=self.mock_end_filter,
+            json=True,
+        )
+
+    def _test_get_workflow_instances_with_status_filter(
+        self, filter_argument: str, status: str, json: bool
+    ):
+        """Tests that the 'get workflow-instances' command works correctly
+        (while filtering based on some status)"""
+
+        self._test_get_workflow_instances_with_filters(
+            filter_arguments=[filter_argument],
+            expected_filters=[self.mock_status_filter.return_value],
+            json=json,
+        )
+        self.mock_status_filter.assert_called_once_with(status)
+
+    # The following uncommented tests each tests that the
```

Comment:
It might be better to have all of these tests in a for loop for all the possible statuses (i.e. take the status names, change it to all lower case + insert into the test title and args)

Reply:
Yeah the only downside is it would be multiple tests in one so if one fails its a little bit harder to see which did, but they are all so similar I think it's ok in this case.

Reply:
I think because of this it may be clearer to leave it as is, unless I added pytest back to use parameterise.